pkg/aws/ec2.go

```diff
@@ -252,9 +252,32 @@ func (r AwsEc2DynamicConfig) SshUser() string {
 	return "ec2-user"
 }
 
-// TODO: implement this function.
+// GetState returns ec's VM state from AWS. See https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceState.html for
+// valid states.
 func (ec AwsEc2DynamicConfig) GetState(kubeClient client.Client, ctx context.Context, instanceId cloud.InstanceIdentifier) (string, error) {
-	return "ACTIVE", nil
+	log := logr.FromContextOrDiscard(ctx)
+	log.Info(fmt.Sprintf("Attempting to get AWS EC2 instance %s's IP address", instanceId))
```

Comment:
alternatively, you can use the following format. It would easier to filter logs

```suggestion
	log.Info("Attempting to get AWS EC2 instance's IP address", "instanceId", instanceId)
```